### PR TITLE
Fixes #32972: Drop RegistryResource supporting crane

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: app/lib/katello/resources/registry.rb [hunk 1]
```diff
@@ -26,18 +26,12 @@ def load_class
...
-
-              # Assume the registry uses the same CA as the Smart Proxy
-              ca_cert_file = Setting[:ssl_ca_file]
-            elsif (container_config = SETTINGS.dig(:katello, :container_image_registry))
-              registry_url = container_config[:crane_url]
-              ca_cert_file = container_config[:registry_ca_cert_file]
-            end
+            uri = URI(pulp_primary.setting(SmartProxy::PULP3_FEATURE, 'content_app_url'))
+            uri.path = "/pulpcore_registry/"
+            registry_url = uri.to_s
```
Comment (ekohl) [inline app/lib/katello/resources/registry.rb:31]: Why is this needed? On line 38 it's parsed again and a new `uri` object created. What's the point?

File: app/lib/katello/resources/registry.rb [hunk 2]
```diff
@@ -26,18 +26,12 @@ def load_class
...
+
+            # Assume the registry uses the same CA as the Smart Proxy
+            ca_cert_file = Setting[:ssl_ca_file]
 
             fail Errors::ContainerRegistryNotConfigured unless registry_url
```
Comment (ekohl) [inline app/lib/katello/resources/registry.rb:36]: Is this still reachable code? I don't think it is. I think it should fail `unless pulp_primary.setting(SmartProxy::PULP3_FEATURE, 'content_app_url')` is set.

**Review by ekohl — CHANGES_REQUESTED** [review]
Yay for cleaning up, but I think you can cleanup a bit more.

**Review by ekohl — APPROVED** [review]
Now we're getting somewhere :)

**Comment by jlsherrill** [comment]
code looks good, let me do a quick test

**Review by jlsherrill — APPROVED** [review]

**Pull request merged**